### PR TITLE
[Constraint solver] Fix backward trailing closures with ".member" expressions

### PR DESCRIPTION
**Pull request by DougGregor**
The introduction of forward-scan matching for trailing closures
(SE-0286) failed to account for unresolved member expressions,
sometimes causing a crash in SILGen. Fixes rdar://problem/67781123.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci smoke test

**Comment by DougGregor** [comment]
@swift-ci smoke test

File: lib/Sema/CSApply.cpp [hunk 1]
```diff
@@ -5592,7 +5592,9 @@ Expr *ExprRewriter::coerceCallArguments(
     SmallVector<LocatorPathElt, 4> path;
     auto anchor = locator.getLocatorParts(path);
     if (!path.empty() && path.back().is<LocatorPathElt::ApplyArgument>() &&
-        (anchor.isExpr(ExprKind::Call) || anchor.isExpr(ExprKind::Subscript))) {
+        (anchor.isExpr(ExprKind::Call) ||
+         anchor.isExpr(ExprKind::Subscript) ||
+         anchor.isExpr(ExprKind::UnresolvedMember))) {
```
[the remark this answers is not in the document]
Comment (DougGregor) [reply lib/Sema/CSApply.cpp]: Second commit improves on this

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]

**Pull request merged**